Clarify GLA header tags

layoutd opened the pull request:
### Changes proposed in this Pull Request:

Closes #813.

- Adds an HTML comment before the Site Verification `<meta>` tag to indicate that it was inserted by Google Listings & Ads.
- Updates the HTML comment before the global site tag `<script>` snippet, also to indicate that it was inserted by Google Listings & Ads.
- Removes unnecessary indentation from the global site tag snippet.

### Detailed test instructions:

1. Connect Merchant Center and Ads accounts.
2. Visit the public-facing store.
3. View the page source code and confirm:
    - The `google-site-verification` meta tag is preceded by the new HTML comment:
    `<!-- Google site verification - Google Listings & Ads -->`
    - The global site tag snippet's HTML comment also indicates that it belongs to GLA:
    `<!-- Global site tag (gtag.js) - Google Ads: AW-8675309 - Google Listings & Ads -->`
    _(Note: won't display if WooCommerce Google Analytics Integration plugin is activated)_

### Changelog entry

> Tweak - Clarify which HTML header tags are inserted by GLA.

layoutd committed:
HTML code comment for site verification meta tag

layoutd committed:
Add extension name to gtag code comment, too.

findingsimple commented:
> Tweak - Clarify which HTML header tags are inserted by GLA>

Genuine question, is the `>` needed on the end of the changelog entry?

mikkamp reviewed (approved):
Works as expected for me. Although it didn't look so bad indented, and we might want to add an extra newline before the gtag comment:
![image](https://user-images.githubusercontent.com/11388669/123385752-ce1c5380-d58d-11eb-85d0-d4442e92207d.png)

layoutd commented:
> Genuine question, is the > needed on the end of the changelog entry?

Genuine answer: only if you want to confuse people 🤦🏻 . Nice catch 🦅 👁️

layoutd committed:
Ensure a newline before the global site tag snippet

layoutd commented:
> Works as expected for me. Although it didn't look so bad indented, and we might want to add an extra newline before the gtag comment:

Thanks @mikkamp. I removed the indentation because I saw most tags in the `<head>` weren't indented (at least on my install). I've added the newline in https://github.com/woocommerce/google-listings-and-ads/pull/825/commits/c3fba475f173b15c04f7265f05f0641414e7d87f, in some instances I see an extra blank line, but better two than none.
